Starting AI/ML section in python docs [New]

xhlulu opened the pull request:
# THIS WAS MOVED TO https://github.com/plotly/plotly.py/pull/2702

Continuation of this PR: https://github.com/plotly/plotly.py/pull/2172
Please refer to: https://github.com/plotly/plotly.py/issues/2153

Changes:
- Added scikit-learn to requirements

Todo list:
- [x] Create a ML section
- [x] kNN Classification notebook
- [x] Regression notebook
- [x] ROC notebook
- [x] PCA notebook
- [ ] tSNE/UMAP notebook
- [ ] LIME (follow [this example](https://marcotcr.github.io/lime/tutorials/Tutorial%20-%20images.html))

emmanuelle commented:
Hi @xhlulu would you like a round of review now or rather later if this is still WIP?

xhlulu commented:
Hi @emmanuelle ! I'm currently incorporating some feedback from Nic, then right after that I'm addressing your reviews from the previous PR. I'll ping you when I'm done with the current task stack :)
